substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/sources/ApplicationSourceCache.java

```diff
@@ -107,6 +107,17 @@ private void trySourceRoot(String sourceRoot, boolean fromClassPath) {
                 // try the path as provided
                 File file = sourcePath.toFile();
                 if (file.exists() && file.isDirectory()) {
+                    // see if we have src/main/java or src/java
+                    Path subPath = sourcePath.resolve("main").resolve("java");
+                    file = subPath.toFile();
+                    if (file.exists() && file.isDirectory()) {
+                        sourcePath = subPath;
+                    } else {
+                        subPath = sourcePath.resolve("java");
```

Comment:
aren't you missing another
```
file = subPath.toFile();
```
here?

Reply:
cc @adinn

Reply:
doh! pushed a correction.

Reply:
hehe ... that's why I prefer Files.isDirectory(...), Files.exists(...), ...